### PR TITLE
Added migration instructions for test case json

### PR DESCRIPTION
**Pull request by breml**
Migrate test case files from old to new format using the
`jq` command.

### CODE REVIEW

**Comment by breml** [comment]
@magnusbaeck I used this command to successfully migrate all our existing test cases files to the new format. Thanks for merging #21.

**Comment by matejzero** [comment]
I can confirm, this command successfully migrated our tests as well.

File: README.md [hunk 1]
```diff
@@ -217,6 +217,21 @@ may have the following properties:
...
+## Migrate to testcases
+
+To migrate test case files from the old to the new config format, which uses
+the `testcases` array to keep the fields `input` and `expected` next
+to each other, the following command using [jq](https://stedolan.github.io/jq/)
+could be used (run this command in the directory containing the test case
+files):
+
+```
+for f in `ls -1 *.json`; do jq '{ codec, fields, ignore, testcases:[[.input[]], [.expected[]]] | transpose | map({input: [.[0]], expected: [.[1]]})}' $f > $f.migrated; mv $f.migrated $f; done
```
Comment (magnusbaeck) [inline README.md]: Super nit: How about `jq ... && mv $f.migrated $f` to avoid clobbering the files if jq for whatever reason fails?

Comment (breml) [inline README.md]: You are right, this is better. Updated the PR.

**Pull request merged**